packages/beacon-state-transition/src/cache/epochContext.ts

```diff
@@ -467,6 +499,18 @@ export class EpochContext {
     return (committeesSinceEpochStart + committeeIndex) % ATTESTATION_SUBNET_COUNT;
   }
 
+  getNextEpochBeaconProposer(): ValidatorIndex[] {
+    const nextShuffling = this.nextShuffling;
+    let nextProposers = this.nextEpochProposers.get(nextShuffling.epoch);
+    if (!nextProposers) {
+      nextProposers = computeProposers(this.nextProposerSeed, nextShuffling, this.effectiveBalanceIncrements);
+      // Only keep proposers for one epoch in the future, so clear before setting new value
+      this.nextEpochProposers.clear();
+      this.nextEpochProposers.set(nextShuffling.epoch, nextProposers);
```

Comment:
In a specific `epochCtx` does `nextShuffling.epoch` ever have different values? This .clear, then .set pattern is weird and miss-leading. Maps are meant to hold values of arbitrary keys, but here nextEpochProposers seems to ever hold a single value